gateway fails to start if there is no internet connection

When I was in Hong Kong, I tried starting the gateway standalone, i.e. not connected to the internet.

It failed to start with the following error:
```
Opening database: /home/dhylands/moziot/.mozilla-iot/config/db.sqlite3
2018-10-02 19:16:32.947 HTTP server listening on port 8080
2018-10-02 19:16:32.988 Not checking SHA256SUMS file for zigbee-adapter since a .git directory was detected
2018-10-02 19:16:33.054 DNS config is: dave-work
2018-10-02 19:16:33.055 Service Discovery: changing profile - local domain is now: dave-work
2018-10-02 19:16:33.136 Service Discovery: state changed to: true
Error: send ENETUNREACH 224.0.0.251:5353
    at Object._errnoException (util.js:1022:11)
    at _exceptionWithHostPort (util.js:1044:20)
    at SendWrap.afterSend [as oncomplete] (dgram.js:499:11)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! things-gateway@0.6.0 start: `webpack && node build/gateway.js "-d"`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the things-gateway@0.6.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

--- a/src/mdns-server.js
+++ b/src/mdns-server.js
@@ -52,9 +52,19 @@ function DNSserviceDiscovery() {
                power: this.power};
   const options = {name: this.localName, host: this.localDomain, txt};
 
+  this.handleError = (e) => {
+    console.debug(`mDNS error: ${e}`);
+    setTimeout(() => {
+      if (this.serviceState) {
+        this.dnssdHandle.start();
+      }
+    }, 10000);
+  };
+
   // Initialize our object and make sure it's not started on object creation.
   this.dnssdHandle =
     dnssd.Advertisement(dnssd.tcp('http'), this.port, options);
+  this.dnssdHandle.on('error', this.handleError);
   this.dnssdHandle.stop();
 }
 
@@ -118,6 +128,7 @@ DNSserviceDiscovery.prototype.setLocalDomain = function(localDomain) {
 
       this.dnssdHandle = dnssd.Advertisement(dnssd.tcp('http'),
                                              this.port, options);
+      this.dnssdHandle.on('error', this.handleError);
       this.dnssdHandle.start();
     }
   } else {
@@ -166,6 +177,7 @@ DNSserviceDiscovery.prototype.changeProfile = function(newProfile) {
 
     this.dnssdHandle = dnssd.Advertisement(dnssd.tcp('http'),
                                            this.port, options);
+    this.dnssdHandle.on('error', this.handleError);
 
     // Check to see if the profile should be active or not.
     if (this.serviceState) {

